File: frontend/src/service/attemptService.js
import api from "../../utils/api.js";


const AttemptService = {
    /**
     * Gửi bài làm Quiz của người dùng.
     * @param {object} userAnswerData - Dữ liệu bài làm (quizId, userId, answers: {questionId: userAnswer})
     * @returns {Promise} Promise chứa response từ API (bao gồm điểm số)
     */
    submitQuiz: (userAnswerData) => api.post('/attempts/submit', userAnswerData),

    /**
     * Lấy thông tin chi tiết của một bài làm theo ID.
     * @param {string|number} id - ID của bài làm
     * @returns {Promise} Promise chứa response từ API
     */
    getAttemptById: (id) => api.get(`/attempts/${id}`),

    /**
     * Lấy danh sách tất cả các bài làm của một người dùng cụ thể.
     * @param {string|number} userId - ID của người dùng
     * @returns {Promise} Promise chứa response từ API
     */
    getAttemptsByUserId: (userId) => api.get(`/attempts/user/${userId}`),

    /**
     * Lấy danh sách tất cả các bài làm cho một Quiz cụ thể.
     * @param {string|number} quizId - ID của Quiz
     * @returns {Promise} Promise chứa response từ API
     */
    getAttemptsByQuizId: (quizId) => api.get(`/attempts/quiz/${quizId}`),
};

export default AttemptService;

File: backend/src/main/java/com/dailycodework/quizonline/service/AttemptService.java
package com.dailycodework.quizonline.service;



import com.dailycodework.quizonline.dto.AttemptDTO;
import com.dailycodework.quizonline.dto.UserAnswerDTO;

import java.util.List;

public interface AttemptService {
    AttemptDTO submitQuiz(UserAnswerDTO userAnswerDTO) throws Exception;
    AttemptDTO getAttemptById(Long id) throws Exception;
    List<AttemptDTO> getAttemptsByUserId(Long userId);
    List<AttemptDTO> getAttemptsByQuizId(Long quizId);
}